lib/rubocop/cop/rails/unprocessable_content_status.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Enforces the use of :unprocessable_content instead of :unprocessable_entity
+      # for HTTP status codes, as :unprocessable_entity is deprecated in Rack 3.1.
+      #
+      # @example
+      #   # bad
+      #   render json: { error: "Invalid data" }, status: :unprocessable_entity
+      #   head :unprocessable_entity
+      #
+      #   # good
+      #   render json: { error: "Invalid data" }, status: :unprocessable_content
+      #   head :unprocessable_content
+      #
+      class UnprocessableContentStatus < Base
```

Comment:
I think it would be better to name it `Rails/DeprecatedHttpStatusNames` instead of specializing it to a specific name. This way, it can be extended in the future when new deprecated status names appear.

Additionally, this cop could also check for `payload_too_large`.
https://github.com/rack/rack/blob/v3.2.0/lib/rack/utils.rb#L576-L579

Reply:
Good suggestion! I've updated the cop to be more generally useful. I noticed the `HttpStatus` cop too and took some inspiration for the updated matcher. I also considered whether it made sense to have that cop handle the deprecations, but thought the deprecation may be helpful to have separately, and to not change the isolated focus of that.